src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs

```diff
@@ -340,6 +339,18 @@ internal static void SetNETFrameworkCompatiblityMode(AppDomainSetup setup, IRunC
                 }
             }
         }
-    }
 #endif
+
+        public static IEnumerable<Attribute> GetCustomAttributes(this Assembly assembly, string fullyQualifiedName)
+        {
+            ValidateArg.NotNull(assembly, nameof(assembly));
+            ValidateArg.NotNullOrWhiteSpace(fullyQualifiedName, nameof(fullyQualifiedName));
+
+            var attributes = assembly
+                .GetCustomAttributes()
+                .Where(i => i.GetType().FullName == fullyQualifiedName);
```

Comment:
Would `assembly.GetType(fullName)` be more efficient than `assembly.CustomAttributes()`? The latter will end up loading and iterating over all the custom attributes in an assembly which could potentially also end up loading other assemblies if for instance, base type of the traversed attribute is defined in a different assembly...

Also, if we keep `assembly.GetCustomAttributes()` call then it may be nice to early out as soon as we find the type - consider using `.FirstOrDefault(i => i.GetType().FullName == fullyQualifiedName)` in place of `.Where(i => i.GetType().FullName == fullyQualifiedName)`.

Reply:
We can first try `assembly.GetType(fullyQualifiedName)` and only if we can't load iterate all attributes.

Reply:
That's a good idea! However, it is also possible to have pathological case where someone has our attribute defined in multiple assemblies and some of the references are to the attribute in the current assembly while others are to the external ones (using extern alias). Hopefully, pathological cases like that would be very very rare 😄

Reply:
That is pathological. We can easily say that we don't support it. IMO